### ISSUE TITLE
Sometimes, zeroing the width of mark glyphs does not work

### ISSUE THREAD

**Issue by madig** [googlefonts/glyphsLib#765]
Leaving as a note to self. I ran into this list of glyphs in a project where going Glyphs.app 2.x -> UFO does not zero the width of the glyph, meaning everything is shifted to the right. Need to figure out why.

```python
MARKS = [
    "co-khmer.below",
    "co-khmer.below2",
    "ta-khmer.below",
    "ta-khmer.below2",
    "cha-khmer.below",
    "qa-khmer.below",
    "qe-khmer.below",
    "ta-khmer.below",
    "ta-khmer.below2",
]
```

**Comment by yanone** [comment]
Is glyphsLib supposed to reduce the width of non-spacing marks automatically like Glyphs does?

I was gonna create an issue but found this issue. The mark in this creates an unusually wide gap because of its positive width and Glyphs users have the expectation that it "just works".

<img width="423" alt="Bildschirmfoto 2022-11-01 um 14 51 31" src="https://user-images.githubusercontent.com/175386/199258468-30d028b6-0836-4a3e-b1d2-48ccd5ad5df8.png">

**Comment by anthrotype** [comment]
it might be that the glyph is not classified as nonspacing mark, either because its glyph name is not recognized as such by the glyphdata or the glyph doesn't have the expected category/subCategory overrides

**Comment by yanone** [comment]
Yeah, I was suspecting that. It turns out well indeed once the definitions are set explicity. Thank you.
